testutils/src/main/java/com/microsoft/identity/internal/testutils/TokenCreator.java

```diff
@@ -0,0 +1,167 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.internal.testutils;
+
+import android.util.Base64;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.crypto.MACSigner;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+
+import java.nio.charset.Charset;
+import java.security.SecureRandom;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public class TokenCreator {
+
+    private static final String NAME_CLAIM = "name";
+    private static final String OBJECT_ID_CLAIM = "oid";
+    private static final String PREFERRED_USERNAME_CLAIM = "preferred_username";
+    private static final String TENANT_ID_CLAIM = "tid";
+    private static final String VERSION_CLAIM = "ver";
+
+    private static final String AUDIENCE = "audience-for-testing";
+    private static final String TENANT_ID = "some-tid";
+    private static final String OBJECT_ID = "some-uid";
+    private static final String PREFERRED_USERNAME = "test@test.onmicrosoft.com";
+    private static final String ISSUER = "https://login.onmicrosoftonline.com/test/v2.0";
+    private static final String SUBJECT = "TestSubject";
+    private static final String VERSION = "2.0";
+    private static final String NAME = "test";
+    private static final String UID = "some-uid";
+    private static final String UTID = "some-tid";
+    private static final String ENCODING_UTF8 = "UTF-8";
+
+    private static String createToken(final String issuer,
```

Comment:
is this `acceesToken` if yes rename to `createMockAccessToken` ?

Reply:
It is a generic method for creating token. I can rename to createMockToken